modules/git/command.go

```diff
@@ -148,15 +160,15 @@ func (c *Command) Run(opts *RunOpts) error {
 		cmd.Env = opts.Env
 	}
 
-	cmd.Env = append(
-		cmd.Env,
-		fmt.Sprintf("LC_ALL=%s", DefaultLocale),
-		// avoid prompting for credentials interactively, supported since git v2.3
-		"GIT_TERMINAL_PROMPT=0",
-		// ignore replace references (https://git-scm.com/docs/git-replace)
-		"GIT_NO_REPLACE_OBJECTS=1",
-	)
+	if HomeDir == "" {
+		// TODO: now, some unit test code call the git module directly without initialization, which is incorrect.
+		// at the moment, we just use a temp HomeDir to prevent from conflicting with user's git config
+		// in future, the git module should be initialized first before use.
+		HomeDir = filepath.Join(os.TempDir(), "/gitea-temp-home")
+		log.Warn("Git's HomeDir is empty, the git module is not initialized correctly, using a temp HomeDir (%s) temporarily", HomeDir)
```

Comment:
Why not fix those tests?

Reply:
This PR has changed a lot, I do not want to mix too many things together.

Reply:
ps: even if some test bite us later, there is a warning and can be fixed easliy. 

If these tests must be fixed, I still prefer to fix them in another PR, instead of mixing together.

Reply:
I created a separate issue to make sure the task won't be forgotten.

* https://github.com/go-gitea/gitea/issues/19874